server/api.go

```diff
@@ -516,6 +516,9 @@ func (p *Plugin) getConnected(c *Context, w http.ResponseWriter, r *http.Request
 func (p *Plugin) getUnreads(c *UserContext, w http.ResponseWriter, r *http.Request) {
 	result, err := p.GitlabClient.GetUnreads(c.Ctx, c.GitlabInfo)
 	if err != nil {
+		if strings.Contains(err.Error(), invalidTokenError) {
+			p.handleRevokedToken(c.GitlabInfo)
+		}
```

Comment:
Instead of duplicating this block across the different functions, I'm thinking we should have a function `handleGitlabError(err error)`, which has the above logic. What do you think @MatthewDorner?

Reply:
Yes, I think that would be better.

Reply:
I guess the function would need to take a `*gitlab.UserInfo` as well. Maybe the `handleGitlabError` should be combined with my `handleRevokedToken` as a single function such as `checkAndHandleRevokedToken`?